examples/http-request-interceptor/http_request_interceptor.bal

```diff
@@ -38,15 +38,15 @@ service class RequestInterceptor {
 // can have only the default path.
 listener http:Listener interceptorListener = new (9090);
 
-// Engage interceptors at the service level. Request interceptor services will be executed from
-// head to tail.
-@http:ServiceConfig {
-    // The interceptor pipeline. The base path of the interceptor services is the same as
-    // the target service. Hence, they will be executed only for this particular service.
-    interceptors: [new RequestInterceptor()]
-}
-service / on interceptorListener {
+// Engage interceptors at the service level using `http:InterceptableService`. The base path of the
+// interceptor services is the same as the target service. Hence, they will be executed only for
+// this particular service.
+service http:InterceptableService / on interceptorListener {
 
+    // Creates the interceptor pipeline. Request interceptor services will be executed from head to tail.
```

Comment:
Can we also update with the improvement we did to support single interceptor as the interceptor pipeline?
```suggestion
    // Creates the interceptor pipeline. The function can return a single interceptor or an array of interceptors as the interceptor pipeline. If the interceptor pipeline is an array, then the request interceptor services will be executed from head to tail.
```

Reply:
Can you check the other BBEs as well? (If we have a single interceptor in an array, then we can remove the array)

Reply:
Updated. I dont think we dont need to update other BBEs as they have the listener level initalization and we anyway need to remove those with ongoing listener level interceptor change.